### PR TITLE
OHARA-1387 Delete individual connector from pipeline

### PR DESCRIPTION
**Pull request by eechih**
可以單獨刪除 FtpSource, JdbcSource, FtpSink, HdfsSink, StreamApp
刪除成功後，自動更新 Pipeline

### CODE REVIEW

File: ohara-manager/client/src/components/common/Tabs/Tabs.js [hunk 1]
```diff
@@ -26,15 +26,11 @@ import { white, shadowNormal, radiusNormal } from '../../../theme/variables';
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Tabs/Tabs.js]: You can use a short import path here ☝️  : `theme/variables`;

File: ohara-manager/client/src/components/pages/PipelinePage/PipelineNewPage.js [hunk 2]
```diff
@@ -205,6 +205,11 @@ class PipelineNewPage extends React.Component {
     });
   };
 
+  refreshGraph = () => {
+    const pipelineId = _.get(this.props.match, 'params.pipelineId', null);
+    this.fetchPipeline(pipelineId);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/PipelineNewPage.js]: Let's do a simple check here before calling the API -> `this.fetchPipeline(pipelineId)`

Or if you're pretty sure that the `pipelineId` will always there. Then no need to use lodash then. Maybe just simply do:

```js
const { pipelineId } = this.props.match.params;
this.fetchPipeline(pipelineId);
```

Comment (konekoya) [reply ohara-manager/client/src/components/pages/PipelinePage/PipelineNewPage.js]: Thanks :)

File: ohara-manager/client/src/components/pages/PipelinePage/connectors/Controller/Controller.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  handleDeleteModalClose = () => {
+    this.setState({ isDeleteModalActive: false });
+  };
+
+  handleDeleteClick = e => {
+    e.preventDefault();
+    const { onDelete } = this.props;
+    if (onDelete) onDelete();
+    this.handleDeleteModalClose();
```
[the remark this answers is not in the document]
Comment (eechih) [reply ohara-manager/client/src/components/pages/PipelinePage/connectors/Controller/Controller.js]: 我想調整： 當 Delete confirm button 按下後，不管 delete api 是否成功，都會關閉 modal。
你覺得如何？

**Comment by konekoya** [comment]
This PR is really neat. Please address these comments and let's get this merge in as soon as possible :)

File: ohara-manager/client/src/components/common/Tabs/Tabs.js (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/src/components/common/Tabs/Tabs.js]: No worries, I forget this all the time :)

File: ohara-manager/client/src/components/pages/PipelinePage/PipelineNewPage.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/pages/PipelinePage/PipelineNewPage.js]: ok

File: ohara-manager/client/src/components/pages/PipelinePage/connectors/Controller/Controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/src/components/pages/PipelinePage/connectors/Controller/Controller.js]: You're right! It should work that way.

[the remark this answers is not in the document]
Comment (eechih) [reply ohara-manager/client/src/components/pages/PipelinePage/connectors/Controller/Controller.js]: fixed

**Comment by konekoya** [comment]
I'll need to test this with the updated backend APIs in tomorrow morning.

**Pull request merged**